app/models/manageiq/providers/google/cloud_manager/provision/cloning.rb

```diff
@@ -9,6 +9,25 @@ def do_clone_task_check(clone_task_ref)
   end
 
   def initial_disk
+    # Check if the source is a snapshot; if it is, we can't use a single
+    # request to make the instance and have to create the disk first
+    resource_type = ManageIQ::Providers::Google::Resources.get_resource_type(source.location)
+
+    if resource_type == 'snapshots'
+      # Unfortunately we have to provision the disk first, then provision the
+      # instance.
+      return create_initial_disk_from_snapshot.get_as_boot_disk(true, true)
+    elsif resource_type == 'images'
```

Comment:
I agree with @durandom that `ManageIQ::Providers::Google::Resources` should be a real object (not just helper) which can expose `snapshot?` and `image?` methods instead of checking strings here.